### PR TITLE
fix: blanking preferred org restores global org, not auto-org [IDE-2031]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
## Summary

- **Bug**: Clearing a folder-level preferred org in the config dialog activated auto-org mode instead of reverting to the global org. Root cause: `applyPreferredOrg` derived `orgSetByUser = preferredOrg != ""`, so blanking the field forced `orgSetByUser=false`, which let the auto-determined org take over.
- **Fix**: Preserve `curOrgSetByUser` from config when the incoming `preferredOrg` is blank; only revert to auto-org mode if the global org is also blank (user truly has no org configured at any level).
- **Tests**: Two new regression tests cover the fixed scenario and the edge case where global org is also blank.
- **Hook change**: Demoted `test-integ` and `test-smoke` from required to advisory in the pre-push hook (pact CLI tools not always available locally; smoke tests take 30-90 min).

## Test plan

- [x] `Test_updateFolderConfig_UserSetOrg_BlankedPreferredOrg_UsesGlobalOrg` — fails before fix, passes after
- [x] `Test_updateFolderConfig_UserSetOrg_BlankedPreferredOrg_GlobalAlsoBlank_RevertToAutoOrg` — covers edge case
- [x] `go test ./application/server/... -run Test_updateFolderConfig` all green
- [x] `go test ./internal/types/...` all green
- [ ] Manual: config dialog — clear org dropdown → effective org shown is global org (requires IDE UI)

___

### **PR Type**
Bug fix, Enhancement

___

### **Description**
- Fixes org selection when blanking preferred org.

- Improves error handling in auth link copy.

- Updates pre-push hook test stage requirements.

- Enhances test for root execution scenario.

___

### Diagram Walkthrough

```mermaid
flowchart LR
  internal/types/folder_config.go -- "Fixes Org Selection Logic" --> application/server/configuration_test.go
  application/server/configuration_test.go -- "Adds Regression Tests" --> Fix(Org Selection Bug Fix)
  domain/ide/command/copy_auth_link.go -- "Adds Nil Guard" --> ErrorHandling(Improved Error Handling)
  scripts/check-tests-run.sh -- "Demotes Test Stages" --> CI(Pre-push Hook Update)
  infrastructure/diagnostics/directory_check/checker_test.go -- "Adds Root Skip" --> TestEnhancement(Test Robustness)
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>configuration_test.go</strong><dd><code>Add tests for blanking preferred org scenarios</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

application/server/configuration_test.go

<ul><li>Adds two new tests:<br>   - <br><code>Test_updateFolderConfig_UserSetOrg_BlankedPreferredOrg_RevertToAutoOrg</code>: <br>Verifies that if the global org is also blank when the preferred org <br>is cleared, it correctly reverts to auto-org mode.<br>   - <br><code>Test_updateFolderConfig_UserSetOrg_BlankedPreferredOrg_UsesGlobalOrg</code>: <br>Verifies that when the preferred org is blanked but a global org <br>exists, <code>OrgSetByUser</code> remains true, and the global org is used instead <br>of auto-org.</ul>

</details>

  </td>
  <td><a href="https://github.com/snyk/snyk-ls/pull/1284/changes#diff-e141faafb3f0507956b85e92ac3b584c7c706bd96e7f428c7ef95d0d8b2df564">+52/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>checker_test.go</strong><dd><code>Skip read-only directory test as root</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

infrastructure/diagnostics/directory_check/checker_test.go

<ul><li>Adds a condition to skip <code>Test_CheckDirectory_ReadOnlyDirectory</code> when <br>the test is run as root.<br> <li> Root user bypasses standard file permission checks, making the test <br>unreliable.</ul>

</details>

  </td>
  <td><a href="https://github.com/snyk/snyk-ls/pull/1284/changes#diff-678179d9dc9c3bae2216c65f6e869d924ae6dec02b68b552729b21372daf3c47">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>copy_auth_link.go</strong><dd><code>Guard against nil notifier in copyAuthLink</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

domain/ide/command/copy_auth_link.go

<ul><li>Adds a nil check for <code>cmd.notifier</code> before calling <code>SendError</code>.<br> <li> Prevents a panic when the notifier is nil and clipboard writing fails.</ul>

</details>

  </td>
  <td><a href="https://github.com/snyk/snyk-ls/pull/1284/changes#diff-333553e98adcda3da433908c173828c8b2ed28aebd61538646479b4ff24d95d7">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>folder_config.go</strong><dd><code>Correctly handle blank preferred org to use global org</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/types/folder_config.go

<ul><li>Modifies the <code>applyPreferredOrg</code> logic to correctly handle blanking the <br>preferred organization.<br> <li> Preserves <code>curOrgSetByUser</code> when <code>preferredOrg</code> is blank, defaulting to <br>the global org.<br> <li> Only reverts to auto-org mode (<code>orgSetByUser = false</code>) if both the <br>preferred and global orgs are blank.<br> <li> Clarifies logic using <code>GetGlobalString</code> for phase-2 lookup.</ul>

</details>

  </td>
  <td><a href="https://github.com/snyk/snyk-ls/pull/1284/changes#diff-74d6e30f4ca44c7cdb655e9b89d7166a3f22765b35a6d94e71574c8620958a65">+9/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>check-tests-run.sh</strong><dd><code>Demote test-integ and test-smoke to advisory</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

scripts/check-tests-run.sh

<ul><li>Demotes <code>test-integ</code> and <code>test-smoke</code> from required stages to advisory <br>stages in the pre-push hook.<br> <li> This change prevents these slow tests from blocking Git pushes if they <br>haven't been run.<br> <li> Updates the logic for displaying advisory warnings to be more generic.</ul>

</details>

  </td>
  <td><a href="https://github.com/snyk/snyk-ls/pull/1284/changes#diff-73ba738421dae7ee6ae3e0fd9175ffaba58b7be97904d5ce0c0dc5e36f1120dd">+4/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: scripts/check-tests-run.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@
 set -e
 
 HASH_FILE=".tests-hash"
-REQUIRED_STAGES=("test" "test-integ" "test-smoke")   # block push if stale
-ADVISORY_STAGES=()                                   # warn only — takes 30-90 min
+REQUIRED_STAGES=("test")                                          # block push if stale
+ADVISORY_STAGES=("test-integ" "test-smoke")                       # warn only — takes 30-90 min
```
Comment (rrama) [inline scripts/check-tests-run.sh:12]: IMO integration tests are fast enough, just the smoke tests are super slow.

**Review by rrama — APPROVED** [review]
